examples/src/ThemeExample.js

```diff
@@ -0,0 +1,43 @@
+// @flow
+import React from 'react';
+import styled, { useTheme, ThemeProvider } from 'styled-components';
+import merge from 'lodash/merge';
+
+import BoxButton from '../../src/components/button';
+import { generateContrastColors } from '../../src/components/theme';
+
+const Button = styled(BoxButton)`
+    background: ${props => props.theme.primary.background};
+    border-color: ${props => props.theme.primary.border};
+    color: ${props => props.theme.primary.foreground};
+
+    .btn:not(.is-disabled)&:hover {
+        background: ${props => props.theme.primary.backgroundHover};
+        border-color: ${props => props.theme.primary.borderHover};
+    }
+
+    .btn:not(.is-disabled)&:active {
+        background: ${props => props.theme.primary.backgroundActive};
+        border-color: ${props => props.theme.primary.borderActive};
+    }
+`;
+
+type Props = {
+    colorKey: ?String,
+};
+
+const ThemeExample = ({ colorKey }: Props) => {
+    let dynamicTheme = {};
```

Comment:
Looks like this can be simplified a bit:

```
const dynamicTheme = colorKey ? generateContrastColors(colorKey) : {};
const theme = merge({}, useTheme(), dynamicTheme);
```